### PR TITLE
kissfft: 131 -> 131.1.0

### PR DESCRIPTION
**Pull request by OPNA2608**
###### Motivation for this change
- Version bump.
- Added more options
  - data type
  - static or shared library
  - with or without OpenMP
  - with or without tools
- Enabled tests

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by OPNA2608** [comment]
Tests fail on x86_64-darwin with a dylib loading error, looking into it.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119758` at 84f29a66 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>ardour_5</li>
    <li>kissfft</li>
    <li>qm-dsp</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/development/libraries/kissfft/default.nix:4:3:
```
  |
4 | , fetchpatch
  |   ^
```
</li>
  </ul>
</details>

**Comment by OPNA2608** [comment]
Result of `nixpkgs-review pr 119758` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>ardour_5</li>
    <li>kissfft</li>
    <li>qm-dsp</li>
  </ul>
</details>

---

Result of `nixpkgs-review pr 119758` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>ardour_5</li>
    <li>kissfft</li>
    <li>qm-dsp</li>
  </ul>
</details>

---

Result of `nixpkgs-review pr 119758` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 packages built:</summary>
  <ul>
    <li>kissfft</li>
  </ul>
</details>

---

Tested on all 3 platforms:

```sh
$ nix-build -E '
with import ./. { };

kissfft.override {
  datatype = "float";
  withTools = true;
  enableOpenmp = true;
}
'
$ ./result/bin/fastconv-float-openmp
# Doesn't crash
$ cat ./result/lib/pkgconfig/kissfft.pc
# Looks fine
```

File: pkgs/development/libraries/kissfft/default.nix [hunk 1]
```diff
@@ -1,45 +1,81 @@
-{ lib, stdenv
+{ lib
+, stdenv
 , fetchFromGitHub
 , fetchpatch
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/kissfft/default.nix:4]:
```suggestion
```

**Review by SuperSandro2000 — APPROVED** [review]
diff LGTM

File: pkgs/development/libraries/kissfft/default.nix (hunk 1 above)
Comment (OPNA2608) [reply pkgs/development/libraries/kissfft/default.nix]: Done.

**Comment by risicle** [comment]
Nice work. WFM macos 10.14.

**Review by risicle — APPROVED** [review]
WFM nixos x86_64 too.

**Pull request merged**